lib/databases/timescale/lootboxes/rewards.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2023 Fluidity Money. All rights reserved. Use of this
+// source code is governed by a GPL-style license that can be found in the
+// LICENSE.md file.
+
+package lootboxes
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/fluidity-money/fluidity-app/lib/log"
+	"github.com/fluidity-money/fluidity-app/lib/timescale"
+)
+
+// UserLootboxCount to group an address and their respective lootboxes
+type UserLootboxCount struct {
+	Address string `json:"address"`
+	LootboxCount float64 `json:"lootbox_count"`
+}
+
+// InsertTopUserReward to insert lootboxes for the given users for their activity during the airdrop.
+// Expects 5 users to reward
+func InsertTopUserReward(currentTime time.Time, users []UserLootboxCount) {
+    if len(users) != 5 {
+        log.Fatal(func(k *log.Log) {
+            k.Format(
+                "Expected top 5 leaderboard winners, but got %d values!",
+                len(users),
+            )
+        })
+    }
+
+	timescaleClient := timescale.Client()
+
+	statementText := fmt.Sprintf(
+		`INSERT INTO %s (
+			address, 
+			transaction_hash, 
+			source, 
+			awarded_time, 
+			volume, 
+			reward_tier, 
+			lootbox_count, 
+			application
+		) VALUES
+		('$1', '', 'leaderboard_prize', '$6', 0, 1, 30, 'none'),
+		('$1', '', 'leaderboard_prize', '$6', 0, 2, 10, 'none'),
+		('$1', '', 'leaderboard_prize', '$6', 0, 3, 5, 'none'),
+
+		('$2', '', 'leaderboard_prize', '$6', 0, 1, 20, 'none'),
+		('$2', '', 'leaderboard_prize', '$6', 0, 2, 5, 'none'),
+		('$2', '', 'leaderboard_prize', '$6', 0, 3, 2, 'none'),
+
+		('$3', '', 'leaderboard_prize', '$6', 0, 1, 15, 'none'),
+		('$3', '', 'leaderboard_prize', '$6', 0, 2, 3, 'none'),
+		('$3', '', 'leaderboard_prize', '$6', 0, 3, 1, 'none'),
+
+		('$4', '', 'leaderboard_prize', '$6', 0, 1, 12, 'none'),
+		('$4', '', 'leaderboard_prize', '$6', 0, 2, 1, 'none'),
+
+		('$5', '', 'leaderboard_prize', '$6', 0, 1, 10, 'none')`,
+		TableLootboxes,
+	)
+
+	_, err := timescaleClient.Exec(
+		statementText,
+		users[0].Address,
+		users[1].Address,
+		users[2].Address,
+		users[3].Address,
+		users[4].Address,
+		currentTime,
+	)
+
+	if err != nil {
+		log.Fatal(func(k *log.Log) {
+			k.Context = Context
+			k.Message = "Failed to insert top user rewards!"
+			k.Payload = err
+		})
+	}
+}
+
+// fetch the 10 addresses with the highest lootboxes earned during the given period
+func GetTopUsersByLootboxCount(startTime, endTime time.Time) []UserLootboxCount {
+	timescaleClient := timescale.Client()
+
+	statementText := fmt.Sprintf(
+		`SELECT 
+			address, 
+			SUM(lootbox_count) AS lootbox_count 
+		FROM %s
+		WHERE 
+			awarded_time >= $1 AT TIME ZONE 'Australia/Adelaide' AND
```

Comment:
Is this the first instance of us doing this explicitly in the database code? Curious how this will work given our lack of it